Document Egress mTLS origination at sidecar using credentialName in DR (#11301)
istio/istio.io: content/en/docs/tasks/traffic-management/egress/egress-tls-origination/index.md

The _SNI_ field is sent unencrypted during the TLS handshake. Using HTTPS prevents the attackers from knowing specific

> topics and articles but does not prevent an attackers from learning that `edition.cnn.com` is accessed.

Reviewer: prevent an attackers --> prevent attackers
It is not your fault. Please also help to fix it.

> topics and articles but does not prevent an attackers from learning that `edition.cnn.com` is accessed.

Reviewer: prevent an attackers --> prevent attackers
it is not your fault...

1. Create required `RBAC` to make sure the secret created in the above step is accessible to the client pod, which is `sleep` in this case.

    {{< text bash >}}
    $ kubectl create role client-credential-role --resource=secret --verb=get,list,watch
    $ kubectl create rolebinding client-credential-role-binding --role=client-credential-role --serviceaccount=default:sleep
    {{< /text >}}

### Configure mutual TLS origination for egress traffic at sidecar

1.  Add a `DestinationRule` to perform mutual TLS origination

    {{< text bash >}}
    $ kubectl apply -f - <<EOF
    apiVersion: networking.istio.io/v1alpha3
    kind: DestinationRule
    metadata:
      name: originate-mtls-for-nginx
    spec:
      workloadSelector:
        matchLabels:
          app: sleep
      host: my-nginx.mesh-external.svc.cluster.local
      trafficPolicy:
        loadBalancer:
          simple: ROUND_ROBIN
        portLevelSettings:
        - port:
            number: 443
          tls:
            mode: MUTUAL
            credentialName: client-credential # this must match the secret created earlier to hold client certs, and works only when DR has a workloadSelector

>             sni: my-nginx.mesh-external.svc.cluster.local

Reviewer: sni is optional here.
you can remove it or add a comment here to avoid possible confusing.

The _SNI_ field is sent unencrypted during the TLS handshake. Using HTTPS prevents the attackers from knowing specific
topics and articles but does not prevent an attackers from learning that `edition.cnn.com` is accessed.

### Cleanup the TLS origination example

Remove the Istio configuration items you created:

    {{< text bash >}}
    $ kubectl delete serviceentry edition-cnn-com
    $ kubectl delete destinationrule edition-cnn-com
    {{< /text >}}

## Mutual TLS origination for egress traffic

This section describes how to configure a sidecar to perform TLS origination for an external service, this time using a
service that requires mutual TLS. This needs the following steps, some of them are similar to the steps mentioned in

> [Egress Gateway TLS Origination](/docs/tasks/traffic-management/egress/egress-gateway-tls-origination)

Reviewer 2: This example is considerably more involved because it requires the following setup:

1. Generate client and server certificates
1. Deploy an external service that supports the mutual TLS protocol
1. Configure client(sleep pod) to use the credentials created in Step 1

> Only then can you configure the external traffic to go through the sidecar which will now perform

Reviewer 2: ```suggestion
Once this setup is complete, you can then configure the external traffic to go through the sidecar which will perform
```

TLS origination.

### Generate client and server certificates and keys

Follow the same steps as [Egress Gateway TLS Origination](/docs/tasks/traffic-management/egress/egress-gateway-tls-origination#generate-client-and-server-certificates-and-keys)

### Deploy a mutual TLS server

> Follow the same steps as [Egress Gateway TLS Origination](/docs/tasks/traffic-management/egress/egress-gateway-tls-origination#deploy-a-mutual-TLS-server)

Reviewer 2: ```suggestion
Follow [these steps](/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/#deploy-a-mutual-tls-server) in the Egress Gateway TLS Origination task.
```

> ### Configure mutual TLS origination for egress traffic

Reviewer 2: ```suggestion
### Configure the client (sleep pod)
```

Author: Done

1.  Create Kubernetes [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to hold the client's certificates:

    {{< text bash >}}
    $ kubectl create secret generic client-credential --from-file=tls.key=client.example.com.key \
      --from-file=tls.crt=client.example.com.crt --from-file=ca.crt=example.com.crt
    {{< /text >}}

    The secret **must** be created in the same namespace as the client pod is deployed in, `default` in this case.

1. Create required `RBAC` to make sure the secret created in the above step is accessible to the client pod, which is `sleep` in this case.

    {{< text bash >}}
    $ kubectl create role client-credential-role --resource=secret --verb=get,list,watch
    $ kubectl create rolebinding client-credential-role-binding --role=client-credential-role --serviceaccount=default:sleep
    {{< /text >}}

> 1.  Create a `ServiceEntry` to enable access to `my-nginx.mesh-external.svc.cluster.local`:

Reviewer 2: ```suggestion
### Configure mutual TLS origination for egress traffic

1.  Create a `ServiceEntry` to enable access to `my-nginx.mesh-external.svc.cluster.local`:
```

    $ kubectl delete destinationrule edition-cnn-com
    {{< /text >}}

## Mutual TLS origination for egress traffic

> Similar to the previous section, this section describes how to configure a sidecar to perform

Reviewer 2: Seems like a lot of duplicate material from the egress gateway task. I would suggest trying to get the common part into a single section (either in this doc, or separate), and then change the other doc (or both) to just say something like: follow [these instruction](link) to set up an external mutual TLS server for the example.
